### PR TITLE
feat(java-binding): generate jni method signature with macro

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

In this PR, we implement a macro `gen_jni_sig` that generates the jni signature from the java method signature. With this macro, we can avoid manually write the list of jni native and signature list when we register native method.

Besides, since we provide a register native function, in the java binding dynamic library, we don't have to export each jni method one by one. We change to provide a single method `JNI_OnLoad`, which will be called and register the jni method when the dynamic library is loaded.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/jni_core/src/lib.rs [hunk 1]
```diff
@@ -296,30 +298,44 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_vnodeCount(
...
 #[no_mangle]
 pub extern "system" fn Java_com_risingwave_java_binding_Binding_hummockIteratorNew<'a>(
     env: EnvParam<'a>,
     read_plan: JByteArray<'a>,
 ) -> Pointer<'static, HummockJavaBindingIterator> {
     execute_and_catch(env, move |env| {
-        let read_plan = Message::decode(to_guarded_slice(&read_plan, env)?.deref())?;
-        let iter = RUNTIME.block_on(HummockJavaBindingIterator::new(read_plan))?;
-        Ok(iter.into())
+        #[cfg(madsim)]
```
Comment (BugenZhao) [inline src/jni_core/src/lib.rs:307]: Does [`#[cfg_or_panic(not(madsim))]`](https://docs.rs/cfg-or-panic/0.2.0/cfg_or_panic/attr.cfg_or_panic.html#) work here?

Comment (wenym1) [reply src/jni_core/src/lib.rs]: Great to know that there is such a useful macro. Thanks for the suggestion!

**Review by chenzl25 — APPROVED** [review]
LGTM! These macros are amazing! Maybe we can remove 
the prefix `Java_com_risingwave_java_binding_Binding_` so that we can have the same native method names in Rust.

**Comment by BugenZhao** [comment]
> LGTM! These macros are amazing! Maybe we can remove the prefix `Java_com_risingwave_java_binding_Binding_` so that we can have the same native method names in Rust.

Find a crate for this -> https://crates.io/crates/jni_fn

**Review by BugenZhao — APPROVED** [review]
LGTM!

**Comment by wenym1** [comment]
> Find a crate for this -> https://crates.io/crates/jni_fn

Just had a look at the crate. It aims to convert the method name to a name that is able to automatically recognized by jvm when loading the dynamic library. However, in this PR, we have changed to register the native methods manually instead of via a predefined name mapping rule, and therefore it's unnecessary for us to maintain the method name according to the rule with the macro in the crate.

**Pull request merged**